bp_common/src/include/bp_common_cache_service_if.vh

```diff
@@ -100,18 +100,19 @@ typedef enum logic [1:0] {
 `define bp_cache_stat_mem_opcode_width $bits(bp_cache_stat_mem_opcode_e)
 
 // data mem pkt structure
-`define declare_bp_cache_data_mem_pkt_s(sets_mp, ways_mp, data_width_mp, cache_name_mp) \
-  typedef struct packed                                                                 \
-  {                                                                                     \
-    logic [`BSG_SAFE_CLOG2(sets_mp)-1:0]      index;                                    \
-    logic [`BSG_SAFE_CLOG2(ways_mp)-1:0]      way_id;                                   \
-    logic [data_width_mp-1:0]                 data;                                     \
-    bp_cache_data_mem_opcode_e                opcode;                                   \
+`define declare_bp_cache_data_mem_pkt_s(sets_mp, ways_mp, block_data_width_mp, fill_width_mp, cache_name_mp) \
+  typedef struct packed                                                                     \
+  {                                                                                         \
+    logic [`BSG_SAFE_CLOG2(sets_mp)-1:0]          index;                                    \
+    logic [`BSG_SAFE_CLOG2(ways_mp)-1:0]          way_id;                                   \
+    logic [fill_width_mp-1:0]                     data;                                     \
+    logic [block_data_width_mp/fill_width_mp-1:0] fill_index;                                \
```

Comment:
Add parentheses around division for clarity on order of operations.

Reply:
Add additional check in nonsynth_if_verif that block_data_width_mp divides evenly by fill_width_mp. I'm assuming we won't want to support a fill width of, e.g., 40-bits when the block width is 512-bits, right? In other words, we will support fill width of like 16, 32, or 64-bits and block widths of 128, 256, 512, which gives us a nice integer ratio of block width : fill width.